runtime/src/accounts_db.rs

```diff
@@ -3243,6 +3243,147 @@ impl AccountsDb {
         total_accounts_after_shrink
     }
 
+    pub(crate) fn minimize_accounts_db(
+        &self,
+        snapshot_slot: Slot,
+        minimized_account_set: &HashSet<Pubkey>,
+    ) {
+        let minimized_slot_set = self.get_minimized_slot_set(minimized_account_set);
+
+        let mut dead_slots = Vec::new();
+        let mut dead_storages = Vec::new();
+        for storages in self.get_snapshot_storages(snapshot_slot, None, None).0 {
+            let slot = storages.first().unwrap().slot();
+            if slot == snapshot_slot {
+                continue;
+            }
+            if minimized_slot_set.contains(&slot) {
+                self.filter_storages(storages, minimized_account_set, &mut dead_storages);
+            } else {
+                dead_slots.push(slot);
+            }
+        }
+
+        self.purge_slots_from_cache_and_store(
+            dead_slots.iter(),
+            &self.external_purge_slots_stats,
+            true,
+        );
+        self.drop_or_recycle_stores(dead_storages);
+    }
+
+    fn get_minimized_slot_set(&self, minimized_account_set: &HashSet<Pubkey>) -> HashSet<Slot> {
+        let mut minimized_slot_set = HashSet::new();
+        self.thread_pool_clean.install(|| {
```

Comment:
see other comment about thread pools. I think dashmap is probably a better choice here.